pkg/validator/handler.go

```diff
@@ -26,7 +37,81 @@ var (
 	Deserializer = codecs.UniversalDeserializer()
 )
 
-func Handler(validator Validator) http.HandlerFunc {
+type HandlerFactoryConfig struct {
+	CtrlClient client.Client
+}
+
+type HandlerFactory struct {
+	ctrlClient client.Client
+}
+
+func NewHandlerFactory(config HandlerFactoryConfig) (*HandlerFactory, error) {
+	if config.CtrlClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.CtrlClient must not be empty", config)
+	}
+
+	h := &HandlerFactory{
+		ctrlClient: config.CtrlClient,
+	}
+
+	return h, nil
+}
+
+func (h *HandlerFactory) NewCreateHandler(validator CreateValidator) http.HandlerFunc {
```

Comment:
Separate `NewCreateHandler` and `NewUpdateHandler` functions, because due to the second one having to decode one more object, and they call different funcs - `Validate` and `ValidateUpdate` respectively.